drivers/src/ecc384.rs

```diff
@@ -379,9 +382,14 @@ impl Ecc384 {
         let mut sig_result = self.sign_internal(priv_key, data, trng);
         let sig = okmutref(&mut sig_result)?;
 
+        // Verify the signature just created
         let r = self.verify_r(pub_key, data, sig)?;
-        caliptra_cfi_lib::cfi_assert_eq_12_words(&r.0, &sig.r.0);
-        sig_result
+        if r == sig.r {
+            caliptra_cfi_lib::cfi_assert_eq_12_words(&r.0, &sig.r.0);
+            sig_result
+        } else {
+            Err(CaliptraError::DRIVER_ECC384_SIGN_VALIDATION_FAILED)
```

Comment:
I believe this should still be DRIVER_ECC384_KEYGEN_PAIRWISE_CONSISTENCY_FAILURE.

PCT means using the newly generated key to sign something and verify that sig. If that process fails, the PCT fails.

Reply:
So I used this error because my understanding was sign() was just the generic signing function that can be called from a few places. This error seemed like the right match for this case.

One of those uses is that key_pair() generation function. In that function, if this sign call fails, key_pair() returns DRIVER_ECC384_KEYGEN_PAIRWISE_CONSISTENCY_FAILURE instead of directly passing this failure up (see line 270). So you'll never actually see this SIGN_VALIDATION_FAILED error when generating a key pair, but you could see it from other callers.

Reply:
sorry, that was my mistake when reviewing. The diff viewer was skipping lines and I thought this was still part of the key_pair() function. I have no concern here.